### PR TITLE
[236] Upgrade to Ruby 2.7.2

### PR DESCRIPTION
**Pull request by gpeng**
### Context

Publish is running on Ruby 2.6.5

### Changes proposed in this pull request

* Ruby to 2.6.5
* Bundler to 2.1.4

To fix issues building the docker container:

* Yarn to 1.22.5
* nodejs to 12.18.4

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Product Review

### CODE REVIEW

**Comment by gpeng** [comment]
[Bump ruby version](https://trello.com/c/GdzjA1UU/236-bump-ruby-version)

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.6.5
+2.7.2
```
Comment (aliuk2012) [inline .ruby-version:1]: might be a good time to remove this in favour of asdf like we have for registers?

File: .tool-versions [hunk 2]
```diff
@@ -1,3 +1,4 @@
-ruby 2.6.5
-yarn 1.12.3
-nodejs 12.15.0
+ruby 2.7.2
+yarn 1.22.5
+nodejs 12.18.4
```
Comment (aliuk2012) [inline .tool-versions:3]: https://github.com/DFE-Digital/publish-teacher-training/blob/master/.nvmrc was missed. I suggest we drop this in favour of asdf

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "engines": {
-    "node": "12.15.0"
+    "node": "12.18.4"
```
Comment (aliuk2012) [inline package.json:3]:
```suggestion
    "node": "12.x"
```
We don't really care about the minor version

**Review by defong — APPROVED** [review]

**Pull request merged**